framework/Supports/Socials_Feed.php

```diff
@@ -0,0 +1,362 @@
+			$media       = $instagram->getUserMedia( 'self', 50 );
+			$insta_posts = $media->data;
+
+		}
+
+
+		foreach ( $insta_posts as $insta_post ) {
+			//pa($insta_post);
+			$timestamp     = $insta_post->created_time;
+			$this->posts[] = array(
+				'post_title'   => 'Instagram post #' . $insta_post->id,
+				'post_content' => ( ! empty( $insta_post->caption->text ) ) ? $insta_post->caption->text : '&nbsp;',
+				'post_date'    => date( 'Y-m-d H:i:s', $timestamp ),
+				'post_name'    => 'instagram_post_' . $insta_post->id,
+				'post_type'    => 'socials_feed',
+				'tax_input'    => array( 'socials_category' => array( 'instagram' ) ),
+				'meta_fields'  => array(
+					'postmeta_image' => ( ! empty( $insta_post->images->standard_resolution->url ) ) ? $insta_post->images->standard_resolution->url : '',
+					'postmeta_url'   => ( ! empty( $insta_post->link ) ) ? $insta_post->link : '',
+					'postmeta_url'   => ( ! empty( $insta_post->videos ) ) ? $insta_post->videos->standard_resolution->url : '',
+				),
+			);
+		}
+
+		return true;
+
+	}
+
+	/**
+	 * Get posts from Twitter
+	 */
+	public function get_twitter_posts() {
+
+
+		$settings = array(
+			'oauth_access_token'        => Theme_Settings::get( 'twitter_access_token' ),
+			'oauth_access_token_secret' => Theme_Settings::get( 'twitter_access_token_secret' ),
+			'consumer_key'              => Theme_Settings::get( 'twitter_customer_key' ),
+			'consumer_secret'           => Theme_Settings::get( 'twitter_customer_secret' ),
+		);
+
+		$url           = 'https://api.twitter.com/1.1/statuses/user_timeline.json';
+		$getfield      = '?screen_name=' . Theme_Settings::get( 'twitter_user_name' );
+		$requestMethod = 'GET';
+
+		$twitter = new \TwitterAPIExchange( $settings );
+
+		$twitter_posts = $twitter->setGetfield( $getfield )
+		                         ->buildOauth( $url, $requestMethod )
+		                         ->performRequest();
+
+		$twitter_posts = json_decode($twitter_posts);
+
+		foreach ( $twitter_posts as $twitter_post ) {
+			$timestamp     = strtotime( $twitter_post->created_at );
+			$this->posts[] = array(
+				'post_title'   => 'Twitter post #' . $twitter_post->id,
+				'post_content' => ( ! empty( $twitter_post->text ) ) ? $twitter_post->text : '&nbsp;',
+				'post_date'    => date( 'Y-m-d H:i:s', $timestamp ),
+				'post_name'    => 'twitter_post_' . $twitter_post->id,
+				'post_type'    => 'socials_feed',
+				'post_status'  => 'draft',
+				'tax_input'    => array( 'socials_category' => array( 'twitter' ) ),
+				'meta_fields'  => array(
+					'postmeta_image' => ( ! empty( $twitter_post->entities->media[0]->media_url ) ) ? $twitter_post->entities->media[0]->media_url : '',
+					'postmeta_url'   => 'https://twitter/' . $twitter_post->user->name . '/status/' . $twitter_post->id,
+				),
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Insert WP posts
+	 *
+	 * @return bool
+	 */
+	public function insert_posts() {
+
+		$this->get_fb_posts();
```

Comment:
Create new method for get social posts to property posts and add him to construct